src/app/access-requests/features/access-request-field-edit/access-request-field-edit.component.ts

```diff
@@ -71,24 +82,33 @@ export class AccessRequestFieldEditComponent implements OnInit {
     this.field.update(() => this.#defaultValue());
   }
 
-  #getValue = () => {
-    if (
-      this.name() === 'ticket_id' &&
-      this.field() &&
-      (this.field() as string).startsWith(this.#baseTicketUrl)
-    ) {
-      this.field.set((this.field() as string).slice(this.#baseTicketUrl.length));
-    }
-    return this.field();
-  };
-
   edit = () => {
     this.isOpen.set(true);
   };
 
-  changed = () => {
+  /**
+   * Remove any URL prefix from the field value if this is a ticket ID
+   * and set the proper error message if the field is invalid.
+   * @param control - the form control for the field
+   */
+  #handleControl = (control: FormControl) => {
+    if (this.name() === 'ticket_id') {
+      // if the field is prefixed with (parts of) the base ticket URL, remove that prefix
+      const baseUrl = this.#baseTicketUrl;
+      let value = this.field() || '';
+      const i = value.lastIndexOf('/');
+      if (baseUrl && i >= 0 && baseUrl.endsWith(value.substring(0, i + 1))) {
```

Comment:
```suggestion
      if (baseUrl && i >= 0 && baseUrl.startsWith(value.substring(0, i + 1))) {
```
Does not work with endsWith for me.
Also should we not strip the URL when we save rather than when the user pastes it? Otherwise it is a little confusing, but it's not a big deal.